### PR TITLE
Fix dynamic clip error in box coder for TensorRT

### PR DESCRIPTION
**Pull request by grimoire**
Hi
This PR use `Tensor.clamp(0, 1)` instead of `torch.where` to fix the conversion error on two-stage detection models such as Faster-RCNN.

### CODE REVIEW

File: mmdet/core/export/onnx_helper.py [hunk 1]
```diff
@@ -21,19 +21,19 @@ def dynamic_clip_for_onnx(x1, y1, x2, y2, max_shape):
...
-    x1 = torch.where(x1 < zero, zero, x1)
-    y1 = torch.where(y1 < zero, zero, y1)
-    x2 = torch.where(x2 < zero, zero, x2)
-    y2 = torch.where(y2 < zero, zero, y2)
-    # clip by h and w
-    x1 = torch.where(x1 > w, w, x1)
-    y1 = torch.where(y1 > h, h, y1)
-    x2 = torch.where(x2 > w, w, x2)
-    y2 = torch.where(y2 > h, h, y2)
+
```
Comment (RunningLeon) [inline mmdet/core/export/onnx_helper.py:24]: could update docstring in line 10
```
...... we have to use torch.where to workaround.
```

**Review by RunningLeon — APPROVED** [review]
LGTM

**Review by ZwwWayne — APPROVED** [review]

File: mmdet/core/export/onnx_helper.py [hunk 2]
```diff
@@ -21,19 +21,24 @@ def dynamic_clip_for_onnx(x1, y1, x2, y2, max_shape):
...
+    # scale by 1/max_shape
```
Comment (ZwwWayne) [inline mmdet/core/export/onnx_helper.py:25]: Why we must use mask shape to to the scaling here? If it is necessary, we need comments here.

Comment (RunningLeon) [reply mmdet/core/export/onnx_helper.py:25]: The reason is mentioned in the docstring.
```
Since torch.clamp cannot have dynamic `min` and `max`, we scale the
      boxes by 1/max_shape and clamp in the range [0, 1].
```
